File: data/min_XOR/LeetCode_Solutions/Original_Solution.py
class Solution:
    def minOperations(self, nums: List[int], k: int) -> int:
        
        xor_all=0
        for num in nums:
            xor_all^=num

        target_xor = xor_all^k
        total_flips = 0

        while target_xor >0:
            if target_xor&1== 1:
                total_flips+= 1
            target_xor >>=1

        return total_flips

File: data/subarrays_atleast_ktimes/LeetCode_Solutions/Original_Solution.java
class Solution {
  public long countSubarrays(int[] nums, int k) {
    final int n = nums.length;
    int max = 0;
    for (int num : nums) {
      max = Math.max(max, num);
    }

    long count = 0;
    // frequency ==> index
    Map<Integer, Integer> seen = new HashMap<>();
    for (int i = 0, current = 0; i < n; i++) {
      if (nums[i] == max) {
        current++;
        seen.put(current, i);
      }
      if (current >= k) {
        // number of subarray end with nums[i]
        count += seen.get(current - k + 1) + 1;
      }
    }

    return count;
  }
}